lib/identity_cache/query_api.rb

```diff
@@ -400,16 +406,18 @@ def prefetch_one_association(association, records)
     def fetch_recursively_cached_association(ivar_name, association_name) # :nodoc:
       if IdentityCache.should_use_cache?
         ivar_full_name = :"@#{ivar_name}"
-
         assoc = if instance_variable_defined?(ivar_full_name)
           instance_variable_get(ivar_full_name)
         else
           instance_variable_set(ivar_full_name, send(association_name))
         end
-
-        assoc.is_a?(ActiveRecord::Associations::CollectionAssociation) ? assoc.reader : assoc
```

Comment:
This is a weird case.  It doesn't look like `send(association_name)` would return a ActiveRecord::Associations::CollectionAssociation.  I think this is working around an inconsistency that shouldn't be there.  I'm going to open up a pull request to fix this, since it looks like it is causing some confusion.